### ISSUE TITLE
New install fail 

### ISSUE THREAD

**Issue by talionwar** [AzuraCast/AzuraCast#212]
I'm installing the latest version on ubuntu server 16 but I can't continue because the error:
TASK [azuracast-db : Apply All DB Migrations] ***********************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "php /var/azuracast/www/util/cli.php migrations:migrate --no-interaction --allow-no-migration", "delta": "0:00:00.181828", "end": "2017-08-16 22:30:10.421915", "failed": true, "rc": 1, "start": "2017-08-16 22:30:10.240087", "stderr": "\n                                                                               \n  [Doctrine\\DBAL\\Exception\\DriverException]                                    \n  An exception occurred while executing 'CREATE TABLE app_migrations (version  \n   VARCHAR(255) NOT NULL, PRIMARY KEY(version)) DEFAULT CHARACTER SET utf8mb4  \n   COLLATE utf8mb4_unicode_ci ENGINE = InnoDB':                                \n  SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was t  \n  oo long; max key length is 767 bytes                                         \n                                                                               \n\n                                                                               \n  [Doctrine\\DBAL\\Driver\\PDOException]                                          \n  SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was t  \n  oo long; max key length is 767 bytes                                         \n                                                                               \n\n                                                                               \n  [PDOException]                                                               \n  SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was t  \n  oo long; max key length is 767 bytes                                         \n                                                                               \n\nmigrations:migrate [--write-sql] [--dry-run] [--query-time] [--allow-no-migration] [--configuration [CONFIGURATION]] [--db-configuration [DB-CONFIGURATION]] [--] [<version>]", "stderr_lines": ["", "                                                                               ", "  [Doctrine\\DBAL\\Exception\\DriverException]                                    ", "  An exception occurred while executing 'CREATE TABLE app_migrations (version  ", "   VARCHAR(255) NOT NULL, PRIMARY KEY(version)) DEFAULT CHARACTER SET utf8mb4  ", "   COLLATE utf8mb4_unicode_ci ENGINE = InnoDB':                                ", "  SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was t  ", "  oo long; max key length is 767 bytes                                         ", "                                                                               ", "", "                                                                               ", "  [Doctrine\\DBAL\\Driver\\PDOException]                                          ", "  SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was t  ", "  oo long; max key length is 767 bytes                                         ", "                                                                               ", "", "                                                                               ", "  [PDOException]                                                               ", "  SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was t  ", "  oo long; max key length is 767 bytes                                         ", "                                                                               ", "", "migrations:migrate [--write-sql] [--dry-run] [--query-time] [--allow-no-migration] [--configuration [CONFIGURATION]] [--db-configuration [DB-CONFIGURATION]] [--] [<version>]"], "stdout": "Loading configuration from the integration code of your framework (setter).\n                                                                    \n                    Doctrine Database Migrations                    \n                                                                    ", "stdout_lines": ["Loading configuration from the integration code of your framework (setter).", "                                                                    ", "                    Doctrine Database Migrations                    ", "                                                                    "]}

RUNNING HANDLER [azuracast-user : send user notification] ***********************************************************************************************************************************

RUNNING HANDLER [mariadb : send mysql notification] *****************************************************************************************************************************************
        to retry, use: --limit @/var/azuracast/www/util/ansible/deploy.retry

PLAY RECAP **********************************************************************************************************************************************************************************
localhost                  : ok=80   changed=18   unreachable=0    failed=1

**Comment by SlvrEagle23** [comment]
This issue seems to be caused by the currently-unresolved doctrine/migrations#514 issue.

Are you trying to install AzuraCast on Ubuntu 14.04? If so, you might find that 16.04 resolves the issue, as it uses a newer version of MariaDB that should resolve the issue.

I will continue to work on fixing this issue in the meantime.

**Comment by parcle** [comment]
Hi, i have same issue on Server version: 10.0.31-MariaDB-0ubuntu0.16.04.2 Ubuntu 16.04

**Comment by andbou** [comment]
For buster:

Why not propose this install script temporary:

apt-get update
apt-get install -q -y git

mkdir -p /var/azuracast/www
cd /var/azuracast/www
git clone https://github.com/AzuraCast/AzuraCast.git .
git checkout bfc28a00af2c84f41a7a2534ceb61810f0d66d0e

chmod a+x install.sh
./install.sh

git checkout master

chmod a+x update.sh
./update.sh

**Comment by SlvrEagle23** [comment]
@andbou I don't recommend checking out a specific non-current branch of the AzuraCast code, but in this case any version of the code before the UTF8MB4 update should install properly, after which you can update through to the newer code.

I'm working with the Doctrine Migrations team to pull request an update that will resolve the issue on their end.

**Comment by parcle** [comment]
@andbou thanks your workaround permit me to install with success to try the product.

**Comment by talionwar** [comment]
I'm installing on Ubuntu 16.04 and got the error

**Comment by talionwar** [comment]
Thanks for the workaround it works!!

**Comment by SlvrEagle23** [comment]
This issue should now be resolved without the need to check out an older version of the code. Please try it out and let me know if you have any issues.

**Comment by SlvrEagle23** [comment]
I'm reverting the Doctrine update, as it should no longer be necessary; all new (and existing) installations will be updated to MariaDB 10.2 from 10.0, which should resolve the UTF8MB4 issue from the get-go.

**Comment by SlvrEagle23** [comment]
I've confirmed on several new installs that this is no longer a problem, since all new installs will use MariaDB 10.2 (across both traditional and Docker installations), giving us feature parity and removing this as an issue that only affects the traditional installs. Closing for now; if anyone encounters any similar issues, please make a new ticket.

**Comment by mgiannelis** [comment]
I have followed these instructions and attempted every single installation method.
These instructions clearly do not work.
I have attempted every single intrucvtion set seperatly on clean Linux Ubuntu installs.
I know that this works. I had someone I used to call a friend install this for me on my self hosted Virtual Machine using Windows HyperV
Please. Would someone update the instruction set to more clear STEP BY STEP instructions.
I have tried everything I could think of. Even ran the instructions on a different machine.
These simply do not work.
The install also says depending on your internet speed that the who process should only take between 10 to 15 minutes.
The install (Once Finished With Constant errors) takes me over an hour on a 100Mbit Connection.
I also cannot find anyone or anywhere else to get any help with this.
I know that it should be working on my Dl360 server because it has been working before.
I cannot find any help, support or additional documentation to help install Azuracast anywhere.
Where is the support. Where can I find someone to help me?
The application simply will not run after installation! 
Please HELP

**Comment by mgiannelis** [comment]
Can someone PLEASE help me to get this installed and up and running again!
I can give anyone that will help me SSH access to a fresh install on UBUNTU Server! 
There has to be someone out there they can help me.
I have been trying to get this up and running for 4 days straight! Ive spend almost 70 hours trying to get it working and installed. 
I have tried the docker method. The Ansible method.
Followed the instructions here
https://www.azuracast.com/install/ansible.html#installing
https://www.azuracast.com/install/docker.html

Nothing works and the install always fails in some way or another.
I have had it running before on a hyperV virtual machine. Someone else installed it for me.
I killed the install by mistake. Now I cant seem to install it again on my own.
The instructions and documents online are so limited!

I am almost at the point of paying anyone that can get it going for me.
I can provide SSH access to a fresh Ubuntu Install.

Please Help! 



Matthew.Giannelis@gmail.com

**Comment by mgiannelis** [comment]
So.
I think I have found something that has actually been missing from the install instructions.
I found that at the end of the install. Part of the error message seemed to included something about permissions.

The instructions on the auzracast site was meant to be followed easy by anyone.
What I think has been missing the whole time.....

The fact that the install needs to be run as a root user.
So I had to enable to root account in ubuntu which is NOT active by default.
If this fixes my problem. I would suggest that this is added to the instructions.
Will let you know if this solves my problem.
